plugins/generic/pln/PLNPlugin.inc.php

```diff
@@ -175,12 +175,29 @@ function getStyleSheet() {
 	 * @param $settingName string
 	 */
 	function getSetting($journalId,$settingName) {
-
-		// if there isn't a journal_uuid, make one
-		if ($settingName == 'journal_uuid') {
-			$uuid = parent::getSetting($journalId, $settingName);
-			if ($uuid) return $uuid;
-			$this->updateSetting($journalId, $settingName, $this->newUUID());
+                $settingValue = parent::getSetting($journalId, $settingName);
+		if ($settingValue != null) return $settingValue;
+                        
+		// if a setting returns null, populate it with a default
+		switch ($settingName) {
```

Comment:
Logically, it doesn't make sense (to me) for a "get setting" operation to potentially introduce settings. But also there are a few modes of operation that have caused problems before WRT settings specifically. In some situations where getSetting(...) will be overridden to avoid a DB access (e.g. before the system is installed, when the database might not be present). This is the case e.g. for plugins that need to take effect in the installer. I don't think OTOH that there are intersections with this yet, but I'd rather not add another wrinkle if we can help it.